include/avk/graphics_pipeline.hpp

```diff
@@ -16,10 +16,25 @@ namespace avk
 		graphics_pipeline_t& operator=(const graphics_pipeline_t&) = delete;
 		~graphics_pipeline_t() = default;
 
-		[[nodiscard]] avk::renderpass renderpass() const { return mRenderPass; }
-		[[nodiscard]] const avk::renderpass_t& renderpass_reference() const { return mRenderPass.get(); }
-		auto renderpass_handle() const { return mRenderPass->handle(); }
-		auto subpass_id() const { return mSubpassIndex; }
+		[[nodiscard]] auto renderpass() const { return mRenderPass; }
+		// TODO(msakmary) I can also keep the consistent naming aka renderpass_reference() return std::optional<std::reference_wrapper>> I feel like this is way cleaner?
+		[[nodiscard]] auto renderpass_pointer() const -> std::optional<const avk::renderpass_t *>

```

Comment:
Yes, please stick to `renderpass_reference()`! Throughout the frameowork, we have the concept of owned instances (e.g. `renderpass`), and references to instances (e.g. `renderpass_reference`), but we never ever have the concept of a pointer.

Also, I don't like the syntax it creates, like e.g.: `*mPipeline->renderpass_pointer().value()` (the required dereference, eew).

Reply:
Okay, reference_wrapper it is!